### PR TITLE
MT08203 fix

### PR DESCRIPTION
**Pull request by angelosa**
Proposed fix for [MT08203](https://mametesters.org/view.php?id=8203)
imagetek_i4100.cpp: add external pin callbacks, disallow internal irq acks for 5 to 7 lines;
metro.cpp: tie irq line 5 to a vblank source with external pin 0 as irq enable, fixes MT08203;

### CODE REVIEW

File: src/devices/video/imagetek_i4100.cpp [hunk 1]
```diff
@@ -72,7 +72,7 @@
 #include <algorithm>
 
 #define LOG_INT (1 << 1U)
-//#define VERBOSE (LOG_INT)
+#define VERBOSE (LOG_INT)
```
Comment (cuavas) [inline src/devices/video/imagetek_i4100.cpp:75]: Don’t forget to back out the extra logging.

File: src/devices/video/imagetek_i4100.h [hunk 2]
```diff
@@ -59,6 +59,9 @@ class imagetek_i4100_device : public device_t,
 	u32 screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
 	void draw_foreground(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
 	DECLARE_WRITE_LINE_MEMBER(screen_eof);
+	auto ext_ctrl_0_cb() { return m_ext_ctrl_0_cb.bind(); }
+	auto ext_ctrl_1_cb() { return m_ext_ctrl_1_cb.bind(); }
+	auto ext_ctrl_2_cb() { return m_ext_ctrl_2_cb.bind(); }
```
Comment (cuavas) [inline src/devices/video/imagetek_i4100.h:64]: Configuration member functions should be kept together – not mixed with live functions.

File: src/devices/video/imagetek_i4100.cpp [hunk 3]
```diff
@@ -670,31 +681,40 @@ void imagetek_i4100_device::window_w(offs_t offset, uint16_t data, uint16_t mem_
...
- ****************************************************/
+ */
 void imagetek_i4100_device::screen_ctrl_w(offs_t offset, uint16_t data, uint16_t mem_mask)
 {
-	m_layer_tile_select[2] = BIT(data,7);
-	m_layer_tile_select[1] = BIT(data,6);
-	m_layer_tile_select[0] = BIT(data,5);
+	m_ext_ctrl_2_cb(bool(BIT(data, 10)));
+	m_ext_ctrl_1_cb(bool(BIT(data, 9)));
+	m_ext_ctrl_0_cb(bool(BIT(data, 8)));
```
Comment (cuavas) [inline src/devices/video/imagetek_i4100.cpp:708]: Read/write line members use `int`, not `bool` – the cast to `bool` means there’s an additional implicit cast from `bool` to `int`.

File: src/devices/video/imagetek_i4100.h (hunk 2 above)
Comment (angelosa) [reply src/devices/video/imagetek_i4100.h:64]: Was treating screen_eof as a configuration option, but I'll move it anyway.

**Pull request merged**